src/searcher/mod.rs

```diff
@@ -365,22 +366,21 @@ impl Searcher {
         let filter = &query_context.filter;
         query_cursor
             .captures(query, tree.root_node(), slice)
-            .filter_map(|(match_, found_capture_index)| {
-                let found_capture_index = found_capture_index as u32;
-                if found_capture_index != capture_index {
+            .filter_map(|(match_, index_into_query_match_captures)| {
+                let this_capture = &match_.captures[index_into_query_match_captures];
```

Comment:
These changes (up through `let single_captured_node = ...`) are "their own thing"

What I ran into was that my understanding of this (what was being called) `found_capture_index` `usize` being returned by the `.captures()` API was incorrect

Based on a little sanity-checking, it appears to be an index into the `.captures` on the other "match" (`tree_sitter::QueryMatch`) that gets returned

(not a "capture index" for the query as a whole) (maybe the `usize` -> `u32` casting should've been a tip-off?)

But so this should get merged into `master` as its own little PR presumably (along with some test cases of queries with multiple "top-level patterns" (which I guess is the only time this issue was happening?))